app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,18 @@
+class LikesController < ApplicationController
+  before_action :require_login
+  
+  def create
+    @user = current_user
+    @image = Image.find(params[:image_id])
+    @like = @image.likes.create
+    redirect_to [@image.gallery, @image]
+  end
+
+  def destroy
+    @like = Like.find(params[:id])
+    @image = @like.image
+    @like.destroy
+    redirect_to [@image.gallery, @image]
```

Comment:
Same idea here with `redirect_to :back`. Also means you don't need to look up the image. So this can become:

``` rb
def destroy
  like = Like.find(params[:id])
  like.destroy
  redirect_to :back
end
```